contracts/modules/Exec.sol

```diff
@@ -102,54 +104,16 @@ contract Exec is BaseLogic {
     /// @param deferLiquidityChecks List of user accounts to defer liquidity checks for
     /// @return List of operation results
     function batchDispatch(EulerBatchItem[] calldata items, address[] calldata deferLiquidityChecks) public reentrantOK returns (EulerBatchItemResponse[] memory) {
-        address msgSender = unpackTrailingParamMsgSender();
-
-        for (uint i = 0; i < deferLiquidityChecks.length; ++i) {
-            address account = deferLiquidityChecks[i];
-
-            require(accountLookup[account].deferLiquidityStatus == DEFERLIQUIDITY__NONE, "e/batch/reentrancy");
-            accountLookup[account].deferLiquidityStatus = DEFERLIQUIDITY__CLEAN;
-        }
-
-
-        EulerBatchItemResponse[] memory response = new EulerBatchItemResponse[](items.length);
-
-        for (uint i = 0; i < items.length; ++i) {
-            EulerBatchItem calldata item = items[i];
-            address proxyAddr = item.proxyAddr;
-
-            uint32 moduleId = trustedSenders[proxyAddr].moduleId;
-            address moduleImpl = trustedSenders[proxyAddr].moduleImpl;
-
-            require(moduleId != 0, "e/batch/unknown-proxy-addr");
-            require(moduleId <= MAX_EXTERNAL_MODULEID, "e/batch/call-to-internal-module");
-
-            if (moduleImpl == address(0)) moduleImpl = moduleLookup[moduleId];
-            require(moduleImpl != address(0), "e/batch/module-not-installed");
-
-            bytes memory inputWrapped = abi.encodePacked(item.data, uint160(msgSender), uint160(proxyAddr));
-            (bool success, bytes memory result) = moduleImpl.delegatecall(inputWrapped);
-
-            if (success || item.allowError) {
-                EulerBatchItemResponse memory r = response[i];
-                r.success = success;
-                r.result = result;
-            } else {
-                revertBytes(result);
-            }
-        }
-
-
-        for (uint i = 0; i < deferLiquidityChecks.length; ++i) {
-            address account = deferLiquidityChecks[i];
-
-            uint8 status = accountLookup[account].deferLiquidityStatus;
-            accountLookup[account].deferLiquidityStatus = DEFERLIQUIDITY__NONE;
+        return doBatchDispatch(items, deferLiquidityChecks, false);
```

Comment:
Is there much gas savings if we *don't* return this?

* The dapp flow with an EOA won't use this since we have simulate now
* Smart contracts can just call the proxies directly and don't need to bother with batches

Reply:
right, you mentioned it already. It would change the function's behavior for anyone integrating, but it saves 1500 gas, getting rid of I think.